### PR TITLE
Vehicle cooler

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

```SUMMARY: Features "Adds a vehicle-mounted area cooler"```

#### Purpose of change
Adds a part that can cool 3-4 tile radius of indoor/sheltered space down to around 4-5 C to make a walk-in fridge or a refridgerated truck area etc

#### Describe the solution
Makes a new field type - cold air, which operates on the opposite principle to hot air.
Instead of worrying about venting ports and where to redirect the heat exchange waste heat to, I just made it a roof mounted part, an the assumption is it just pumps the hot air out above where it dissipates rapidly ( which it seems a lot of refrigerated trucks do aswell irl )

#### Describe alternatives you've considered
A part that worked like a rimworld cooler, and had a hot side and a cold side, but that would require some unnecessary logic to how the part worked when the vehicle was turning etc.

#### Additional context
N/A

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Good job. Cold air was on my TODO list and I can scratch that now.

Shouldn't cold air and hot air cancel each other out? Also with different levels of hotness/coldness it would need to be gradual. Such interaction might be added to the field processing chunk of the code.

**Comment by davidpwbrown** [comment]
> Good job. Cold air was on my TODO list and I can scratch that now.
> 
> Shouldn't cold air and hot air cancel each other out? Also with different levels of hotness/coldness it would need to be gradual. Such interaction might be added to the field processing chunk of the code.

Mixing different fields and interacting different gases is something none of the other fields and gases do, and could be extremely complex.
Hot air and cold air do cancel each other out when it comes to the temperature calculation.
If someone wants to do simulation of gaseous motion and mixing, then thatd be great, but that would be way beyond me!

**Comment by nexusmrsep** [comment]
I volunteer to do those interactions when this gets merged. The logic behind it would be following: cold/hot air both have effective 4x3=12 levels of intensity. If you compare and substract those levels at a tile with both cold+hot air you can destroy both fields and spawn a new one based on the resulting intensity. If the levels are equal you just destroy both fields effectively cancelling each other out. There might be some logic involved with field's age, but I believe that if a new field would not be older then the max age of interacting old fields it'd fine.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -5501,6 +5506,16 @@ void vehicle::update_time( const time_point &update_to )
         g->m.adjust_field_strength( global_part_pos3( pt ), fd_hot_air3, density );
         discharge_battery( pt.info().epower );
     }
+    // coolers emitting cold air
+    for( int idx : coolers ) {
+        const auto &pt = parts[idx];
```
Comment (ifreund) [inline src/vehicle.cpp]:
```suggestion
        const vehicle_part &pt = parts[idx];
```

**Comment by Shibimon** [comment]
How hard would be to add some kind of regulator to manually choose the temperature this will go for?
It would be a really useful tool to cool places during summer.

**Pull request merged**